src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/ChainedMethodCallWrapCheck.java

```diff
@@ -0,0 +1,448 @@
+ * Default value is {@code "^$"}.
+ * </li>
+ * <li>
+ * Property {@code maxCallsPerLine} - Specify maximum number of method calls per lines if we
+ * are wrapping the method calls. If all the method calls span over a single line then this
+ * property has no effect.
+ * Type is {@code int}.
+ * Default value is {@code 1}.
+ * </li>
+ * <li>
+ * Property {@code maxSingleLineCalls} - Specify maximum number of method calls if all chained
+ * method calls span over a single line. If chained method calls span over multiple lines then
+ * this property has no effect.
+ * Type is {@code int}.
+ * Default value is {@code 1}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check to use custom identifierPattern to validate assertions:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;ChainedMethodCallWrap&quot;&gt;
+ *  &lt;property name=&quot;identifierPattern&quot; value=&quot;^assert.*$&quot; /&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * void method() {
+ *     assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *                      // violation above, ''3' method calls on single line, max allowed is '1''
+ *     assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // violation, 'Chained method call should be wrapped'
+ *
+ *     assertWithSomeMessage("ex")
+ *         .that(1)
+ *         .isEqualTo(1); // ok, every method call is wrapped
+ *
+ *     Truth.assertWithSomeMessage("ex").that(1).isEqualTo(1); // ok, identifierPattern is different
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to use custom identifierPattern to validate Truth's assertions:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;ChainedMethodCallWrap&quot;&gt;
+ *  &lt;property name=&quot;identifierPattern&quot; value=&quot;^Truth\.assert.*$&quot; /&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * void method() {
+ *     Truth.assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *                      // violation above, ''3' method calls on single line, max allowed is '1''
+ *     Truth.assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // violation, 'Chained method call should be wrapped'
+ *
+ *     assertWithSomeMessage("ex").that(1).isEqualTo(1); // ok, identifierPattern is different
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to use custom identifierPattern and maxSingleLinesCalls:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;ChainedMethodCallWrap&quot;&gt;
+ *  &lt;property name=&quot;identifierPattern&quot; value=&quot;^assert.*$&quot; /&gt;
+ *  &lt;property name=&quot;maxSingleLineCalls&quot; value=&quot;3&quot; /&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * void method() {
+ *     assertWithSomeMessage("ex").that(1).isEqualTo(1);
+ *                                // ok above, 3 method calls &lt;= maxSingleLineCalls
+ *     assertWithSomeMessage("ex")
+ *         .that(1).isEqualTo(1); // violation, 'Chained method call should be wrapped'
+ * }
```

Comment:
Please use same code for each example; user wants to change their config, not their code.

Reply:
Done.

Reply:
Resolved.